### DISCUSSION TITLE
Error on compiling Rocket , wondering is this just problem for me

### DISCUSSION THREAD

**Discussion by rajasankar** [rwf2/Rocket#2642] [Questions] [answered]
Hi All, 

Got this error. Is it only for me or others got this too. Just wanted to post here to confirm before posting it as a issue ticket. 

Any one else facing the same issue? 

Compiling rocket v0.5.0-rc.4
error[E0061]: this function takes 1 argument but 0 arguments were supplied
  --> /home/.cargo/registry/src/github.com-1ecc6299db9ec823/rocket-0.5.0-rc.4/src/server.rs:76:5
   |
76 |     debug!("Received request: {:#?}", hyp_req);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ an argument of type `UnsafeArg` is missing
   |
note: associated function defined here
  --> /home/.cargo/registry/src/github.com-1ecc6299db9ec823/yansi-1.0.0-rc.1/src/paint.rs:56:8
   |
56 |     fn new(self) -> Painted<Self> where Self: Sized {
   |        ^^^
   = note: this error originates in the macro `$crate::__private_api::format_args` which comes from the expansion of the macro `debug` (in Nightly builds, run with -Z macro-backtrace for more info)
help: provide the argument
  --> /home/.cargo/registry/src/github.com-1ecc6299db9ec823/log-0.4.20/src/macros.rs:51:62
   |
51 |                 $crate::__private_api::format_args!($($arg)+)(/* UnsafeArg */),
   |                                                              +++++++++++++++++

error[E0308]: mismatched types
  --> /home/.cargo/registry/src/github.com-1ecc6299db9ec823/rocket-0.5.0-rc.4/src/server.rs:76:5
   |
76 |     debug!("Received request: {:#?}", hyp_req);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `UnsafeArg`, found struct `Painted`
   |
   = note: expected struct `UnsafeArg`
              found struct `Painted<UnsafeArg>`
   = note: this error originates in the macro `$crate::__private_api::format_args` which comes from the expansion of the macro `debug` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0061]: this function takes 1 argument but 0 arguments were supplied
   --> /home/.cargo/registry/src/github.com-1ecc6299db9ec823/rocket-0.5.0-rc.4/src/server.rs:163:9
    |
163 |         debug!("sending response: {:#?}", hyp_response);
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ an argument of type `UnsafeArg` is missing
    |
note: associated function defined here
   --> /home/.cargo/registry/src/github.com-1ecc6299db9ec823/yansi-1.0.0-rc.1/src/paint.rs:56:8
    |
56  |     fn new(self) -> Painted<Self> where Self: Sized {
    |        ^^^
    = note: this error originates in the macro `$crate::__private_api::format_args` which comes from the expansion of the macro `debug` (in Nightly builds, run with -Z macro-backtrace for more info)
help: provide the argument
   --> |/home/.cargo/registry/src/github.com-1ecc6299db9ec823/log-0.4.20/src/macros.rs:51:62
    |
51  |                 $crate::__private_api::format_args!($($arg)+)(/* UnsafeArg */),
    |                                                              +++++++++++++++++

error[E0308]: mismatched types
   --> /home/.cargo/registry/src/github.com-1ecc6299db9ec823/rocket-0.5.0-rc.4/src/server.rs:163:9
    |
163 |         debug!("sending response: {:#?}", hyp_response);
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `UnsafeArg`, found struct `Painted`
    |
    = note: expected struct `UnsafeArg`
               found struct `Painted<UnsafeArg>`
    = note: this error originates in the macro `$crate::__private_api::format_args` which comes from the expansion of the macro `debug` (in Nightly builds, run with -Z macro-backtrace for more info)

Some errors have detailed explanations: E0061, E0308.
For more information about an error, try `rustc --explain E0061`.
error: could not compile `rocket` due to 4 previous errors

**Comment by SergioBenitez** [comment, marked as answer]
Someone reported a similar problem on Matrix. It's some sort of issue with `rustc`, it seems: updating/downgrading your toolchain will likely resolve the problem. Please let us know if it does/doesn't.

**Comment by rajasankar** [comment]
Thank you @SergioBenitez  for the immediate reply. I forgot that step. It worked and complied. I will keep that in mind next time. 
Thanks a lot for the awesome framework.
